packages/askar/src/AskarModule.ts

```diff
@@ -16,50 +17,64 @@ export class AskarModule implements Module {
     this.config = new AskarModuleConfig(config)
   }
 
+  public api = AskarApi
+
   public register(dependencyManager: DependencyManager) {
     dependencyManager.registerInstance(AskarModuleConfig, this.config)
 
-    if (dependencyManager.isRegistered(InjectionSymbols.Wallet)) {
-      throw new CredoError('There is an instance of Wallet already registered')
+    if (!this.config.enableKms && !this.config.enableStorage) {
+      dependencyManager
+        .resolve(AgentConfig)
+        .logger.warn(`Both 'enableKms' and 'enableStorage' are disabled, meaning Askar won't be used by the agent.`)
     }
-    dependencyManager.registerContextScoped(InjectionSymbols.Wallet, AskarWallet)
 
-    // If the multiWalletDatabaseScheme is set to ProfilePerWallet, we want to register the AskarProfileWallet
-    if (this.config.multiWalletDatabaseScheme === AskarMultiWalletDatabaseScheme.ProfilePerWallet) {
-      dependencyManager.registerContextScoped(AskarProfileWallet)
+    if (this.config.enableKms) {
+      const kmsConfig = dependencyManager.resolve(Kms.KeyManagementModuleConfig)
+
+      // Register askar backend if not registered yet
+      if (!kmsConfig.backends.find((backend) => backend.backend === AksarKeyManagementService.backend)) {
```

Comment:
Shouldn't we throw error if it is already registered? Do you see cases where the KMS backend could have been registered by another module and not be a mistake by the dev?

Reply:
You can register it manually on the kms module as well. But in that case maybe you should set `enableKms: false` as you register it manually? I agree

Reply:
This is the message now: Unable to register AskarKeyManagementService. There is a key management backend with name '${AskarKeyManagementService.backend}' already registered. If you have manually registered the AskarKeyManagementService on the KeyManagementModule, set 'enableKms' to false in the AskarModule.